controllers/dockermachine_controller.go

```diff
@@ -329,3 +362,37 @@ func (r *DockerMachineReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		).
 		Complete(r)
 }
+
+func (r *DockerMachineReconciler) reconcileDelete(
+	ctx context.Context,
+	cluster *capiv1alpha2.Cluster,
+	machine *capiv1alpha2.Machine,
+	dockerMachine *infrastructurev1alpha2.DockerMachine,
+) (ctrl.Result, error) {
+
+	if isControlPlaneMachine(machine) {
+		err := actions.DeleteControlPlane(cluster.GetName(), machine.GetName())
+		if err != nil {
+			r.Log.Error(err, "Error deleting control plane node", "nodeName", machine.GetName())
+			return ctrl.Result{RequeueAfter: time.Second * 30}, err
+		}
+	} else {
+		err := actions.DeleteWorker(cluster.GetName(), machine.GetName())
+		if err != nil {
+			r.Log.Error(err, "Error deleting worker node", "nodeName", machine.GetName())
+			return ctrl.Result{RequeueAfter: time.Second * 30}, nil
+		}
+	}
+
+	dockerMachine.ObjectMeta.Finalizers = util.Filter(dockerMachine.ObjectMeta.Finalizers, capiv1alpha2.MachineFinalizer)
```

Comment:
This is using the capi machine finalizer but should be using the finalizer defined in this PR